nautobot_ssot/jobs/base.py

```diff
@@ -47,17 +48,56 @@ class DataSyncBaseJob(BaseJob):
 
     dry_run = BooleanVar()
 
+    def load_source_adapter(self):
+        """TODO:."""
+        pass
+
+    def load_target_adapter(self):
+        """TODO:."""
+        pass
+
+    def calculate_diff(self):
+        """TODO:."""
+        pass
+
+    def execute_sync(self):
+        """TODO:."""
+        pass
+
     def sync_data(self):
         """Method to be implemented by data sync concrete Job implementations.
 
+        Note: You could completely overwrite it if desired.
+
         Available instance attributes include:
 
         - self.kwargs     (corresponds to the Job's `data` input, including 'dry_run' option)
         - self.commit     (should generally be True)
         - self.sync       (Sync instance tracking this job execution)
         - self.job_result (as per Job API)
         """
-        pass  # pylint: disable=unnecessary-pass
+        # TODO: most of the log messages from the methods could be standarised here
+        # example: self.log_info(message="Loading current data from source adapter...")
+        start_time = datetime.now()
+        self.load_source_adapter()
+        load_source_adapter_time = datetime.now()
+        self.load_target_adapter()
+        load_target_adapter_time = datetime.now()
+        self.calculate_diff()
+        calculate_diff_time = datetime.now()
+
+        time_message = (
+            f"Remote Load Time: {load_source_adapter_time - start_time}\n",
+            f"Local Load Time: {load_target_adapter_time - load_source_adapter_time}\n",
+            f"Diff Time: {calculate_diff_time - load_target_adapter_time}\n",
+        )
+
+        if not self.kwargs["dry_run"]:
+            self.execute_sync()
+            execute_sync_time = datetime.now()
+            time_message += (f"Sync Time: {execute_sync_time - calculate_diff_time}",)
+
+        self.log_info(message=time_message)
```

Comment:
Personally I'd rather see four separate timing log messages, each at the appropriate point in the logs, instead of one combined message at the end of the logs.